pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/ZKMetadataUtils.java

```diff
@@ -56,6 +62,31 @@ public static void refreshSegmentZKMetadata(String tableNameWithType, SegmentZKM
         segmentSizeInBytes, false);
   }
 
+  public static void updateSegmentZKMetadataInterval(TableConfig tableConfig, SegmentZKMetadata segmentZKMetadata,
```

Comment:
We can get the `DateTimeFieldSpec` on the caller side and pass it in here
```suggestion
  public static void updateSegmentZKTimeInterval(SegmentZKMetadata segmentZKMetadata, DateTimeFieldSpec dateTimeFieldSpec)
```